src/Test/HttpClient/MockHttpClient.php

```diff
@@ -2,21 +2,19 @@
 
 namespace FOS\HttpCache\Test\HttpClient;
 
-use Http\Adapter\Exception\MultiHttpAdapterException;
-use Http\Adapter\HttpAdapter;
-use GuzzleHttp\Psr7\Response;
+use Http\Client\HttpClient;
 use Http\Discovery\MessageFactoryDiscovery;
 use Psr\Http\Message\RequestInterface;
 use Psr\Http\Message\ResponseInterface;
 
 /**
- * HTTP adapter mock
+ * HTTP client mock
  *
  * This mock is most useful in tests. It does not send requests but stores them
  * for later retrieval. Additionally, you can set an exception to test
  * exception handling.
  */
-class MockHttpAdapter implements HttpAdapter
+class MockHttpClient implements HttpClient
```

Comment:
@sagikazarmark could this class go into the php-http/utils ? its a utility to mock a Httplug client for unit testing that is more convenient than prophesize and phpunit mocks.